slackbridge/bots.py

```diff
@@ -129,17 +131,48 @@ def log(self, method, message):
         return method(full_message)
 
     def signedOn(self):
-        # If already registered, auth in
-        self.msg('NickServ', 'IDENTIFY {}'.format(self.nickserv_password))
-        # And if not, register for the first time
-        self.msg('NickServ', 'GROUP {} {}'.format(self.target_group_nick,
-                                                  self.nickserv_password))
+        self.nickserv_auth()
+
         for channel_name in self.joined_channels:
             self.log(log.msg, 'Joining #{}'.format(channel_name))
             self.join(channel_name)
 
         self.away('Default away for startup.')
 
+    def nickserv_auth(self):
+        # If already registered, authenticate yourself to Nickserv
+        self.msg('NickServ', 'IDENTIFY {}'.format(self.nickserv_password))
```

Comment:
I think this should probably only happen when it's the intended nickname too, otherwise it won't be able to identify with the desired nickname anyway. That being said, if it tries and fails, it's not a big deal. We'd probably also want to go through the registered nicknames in nickserv and remove those ones that are aliases right now, since we don't want to keep them.